xgi/utils/tensor.py

```diff
@@ -0,0 +1,191 @@
+## Tensor times same vector in all but one (TTSV1) and all but two (TTSV2)
+from collections import defaultdict
+from itertools import combinations
+from math import factorial
+
+import numpy as np
+from numpy import prod
+from scipy.signal import convolve
+from scipy.sparse import coo_array
+from scipy.special import binom as binomial
+
+__all__ = [
+    "pairwise_incidence",
+    "ttsv1",
+    "ttsv2",
+]
+
+
+def pairwise_incidence(edgedict):
+    """Create pairwise adjacency dictionary from hyperedge list dictionary
+
+    Parameters
+    ----------
+    edgedict : dict
+        edge IDs are keys, edges are values
+
+    Returns
+    -------
+    pairs : dict
+        a dictionary with node pairs as keys and the hyperedges they appear in as values
+    """
+    pairs = defaultdict(set)
+    for e, edge in edgedict.items():
+        for i, j in combinations(sorted(edge), 2):
+            pairs[(i, j)].add(e)
+        for n in edge:
+            pairs[(n, n)].add(e)
+    return pairs
+
+
+def banerjee_coeff(l, r):
+    """Return the Banerjee alpha coefficient
```

Comment:
is this a standard coefficient? googling didn't help me. Either change the name, or say what it relates to, and maybe add a ref?

Reply:
We should add reference to the Banerjee paper https://www.sciencedirect.com/science/article/pii/S0024379516306115?via%3Dihub

Reply:
Reference added.